src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,69 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
+import { UserInfo } from '../UserInfo';
+
+type Props = {
+  setSelectedTodo: (todo: null) => void,
+  selectedTodo: Todo,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  setSelectedTodo,
+  selectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const [error, setError] = useState('');
+
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = selectedTodo;
+
+  const loadUserFromServer = useCallback(async () => {
+    try {
+      const result = await getUser(userId);
+
+      setUser(result);
+      setError('');
+    } catch (err) {
+      if (err instanceof Error) {
+        setError(err.message);
+      }
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    loadUserFromServer();
+  }, []);
+
+  const handleKeyPress = (event: KeyboardEvent) => {
+    if (event.key === 'x') {
+      setSelectedTodo(null);
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleKeyPress);
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyPress);
+    };
+  }, []);
```

Comment:
Add `[handleKeyPress]` to the dependency array of `useEffect` to ensure the effect runs when `handleKeyPress` changes